src/patients/GeneralInformation.tsx

```diff
@@ -209,52 +241,448 @@ const GeneralInformation = (props: Props) => {
+                      />
+                    </td>
+                  </tr>
+                ))}
+              {phoneNumbers?.map(
+                (p: PhoneNumber, index: number) =>
+                  isEditable && (
+                    <tr key={p.id}>
+                      <td>
+                        <SelectWithLabelFormGroup
+                          name="temporaryPhoneNumberType"
+                          label={t('patient.phoneNumber.type')}
+                          value={p.type}
+                          isEditable={isEditable}
+                          options={[
+                            { label: t('patient.phoneNumber.types.home'), value: 'home' },
+                            { label: t('patient.phoneNumber.types.work'), value: 'work' },
+                            { label: t('patient.phoneNumber.types.temporary'), value: 'temporary' },
+                            { label: t('patient.phoneNumber.types.old'), value: 'old' },
+                            { label: t('patient.phoneNumber.types.mobile'), value: 'mobile' },
+                          ]}
+                          onChange={(event) => {
+                            const type = event.target.value
+                            setPhoneNumbers((currentPhoneNumber) =>
+                              produce(currentPhoneNumber, (v) => {
+                                v[index].type = type
+                              }),
+                            )
+                          }}
+                        />
+                      </td>
+                      <td>
+                        <TextInputWithLabelFormGroup
+                          name="temporaryPhoneNumber"
+                          isRequired
+                          label={t('patient.phoneNumber.phoneNumber')}
+                          isEditable={isEditable}
+                          placeholder="PhoneNumber"
+                          value={p.phoneNumber}
+                          onChange={(event) => {
+                            const phoneNumber = event.target.value
+                            setPhoneNumbers((currentPhoneNumber) =>
+                              produce(currentPhoneNumber, (v) => {
+                                v[index].phoneNumber = phoneNumber
+                              }),
+                            )
+                          }}
+                          feedback={t(error?.phoneNumber)}
+                          isInvalid={!!error?.phoneNumber}
+                          type="tel"
+                        />
+                      </td>
+                    </tr>
+                  ),
+              )}
+            </tbody>
+          </table>
+          <div className="addButtonWrapper">
+            {isEditable && (
+              <Button
+                outlined
+                className="addButton"
+                color="success"
+                icon="add"
+                iconLocation="left"
+                onClick={() => {
```

Comment:
When a phone number is entered and user does not click on the "Add" button, and user saves the form, the phone number is not saved. Perhaps, any form entry can be considered as user input to be validated?

Reply:
Similarly, I wonder if we can have a way to delete a phone number by checking if user input is empty or not at the time of validation? Then, we might not need a delete button.